org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/mixins/MixinsTestCase.xtend

```diff
@@ -0,0 +1,369 @@
+package org.uqbar.project.wollok.tests.interpreter.mixins
+
+import org.uqbar.project.wollok.tests.interpreter.AbstractWollokInterpreterTestCase
+import org.junit.Test
+
+/**
+ * 
+ * @author jfernandes
+ */
+class MixinsTestCase extends AbstractWollokInterpreterTestCase {
+	
+	@Test
+	def void simpleMixinWithMethod() {
+		'''
+		mixin Flies {
+			method fly() {
+				console.println("I'm flying")
+			}
+		}
+		class Bird mixed with Flies {}
+		program t {
+			val b = new Bird()
+			b.fly()
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void classWithMixinThatModifiesTheMixinStateFromTheMixin() {
+		'''
+		mixin Walks {
+			var walkedDistance = 0
+			method walk(distance) {
+				walkedDistance += distance
+			}
+			method walkedDistance() = walkedDistance
+		}
+		class WalkingBird mixed with Walks {}
+		
+		program t {
+				val b = new WalkingBird()
+				b.walk(10)
+				assert.equals(10, b.walkedDistance())
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	// is it ok to have 'protected' variables ?
+	// this is inconsistent with the way wollok handles variables in classes in a hierarchy
```

Comment:
Not currently.
Instance variables in classes are private.
But a subclass cannot declare a new one with the same name as one in a superclass.
Although is just a static check.

Mixins instance variables on the other hand was implemented as protected.

I woudln't be a problem to change class variables to protected.